iceoryx_examples/ice_multi_publisher/ice_re_subscriber.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/guard_condition.hpp"
+#include "iceoryx_posh/popo/modern_api/typed_subscriber.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "topic_data.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <iostream>
+
+bool killswitch = false;
+
+static void sigHandler(int sig [[gnu::unused]])
+{
+    killswitch = true;
+}
+
+constexpr uint64_t MAX_HISTORY_SIZE{4};
+constexpr uint64_t UNSUBSCRIBED_TIME_SECONDS{3};
+
+void receive()
+{
+    iox::popo::TypedSubscriber<CounterTopic> subscriber({"CounterTopic", iox::capro::AnyInstanceString, "Counter"});
```

Comment:
This is the same topic. AnyInstanceString is currently only used in an ara::com context. If you use it here we have to clarify the meaning of the three CapRo IDs. In ara::com you also have to create a proxy (with event receiver) for a specific instance. The AnyInstance is only for discovery. I didn't knew that we can do this here at all

Reply:
Ok, I will remove this then I think and just have to publishers sending on the same topic, meaning all 3 capro identifiers are the same.

Reply:
Makes sense. We first have to think about the CapPro Id mapping one more time. Like this it is not compatible to ara::com and I guess also not to rmw_iceoryx. The users have some flexibility how to use the three ID strings but we should not come up with an interpretation that is completely new

Reply:
In general it would be nice to express something like: give me all data from all lidar sensors (instances of a service) in a specific format (essentially the event). This would merge the received data from all instances, i.e. multiple sensors which is useful so the user does not have to do this himself. But yes, maybe confusing for now.

Reply:
Yes, that's cool. Maybe we should also think about renaming the IDs in the CapPro service description. And then define a behavior and functionality of iceoryx and the mapping to ara::com, DDS and ROS2